newsroom/notifications/models.py

```diff
@@ -0,0 +1,34 @@
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Annotated, Optional
+from superdesk.core.resources import ResourceModel
+from superdesk.core.resources.fields import ObjectId, Field
+from superdesk.core.resources.validators import validate_data_relation_async
+from newsroom.users.module import users_resource_config
+
+
+user_validated_type = Annotated[ObjectId, validate_data_relation_async(users_resource_config.name)]
+
+
+class Notification(ResourceModel):
+    item: str | ObjectId
+    # TODO-async: replace the one above with the one below once `items` are migrated to async framework
+    # item: Annotated[Optional[ObjectId], validate_data_relation_async("items")] = None
+
+    user: user_validated_type
+    resource: Optional[str] = ""
+    action: Optional[str] = ""
+    data: Optional[dict] = {}
+
+
+@dataclass
+class Topic:
```

Comment:
There is already a class named Topic, maybe a more meaningful name here would describe what it is used for and be unique.